app/javascript/components/schedule-form/schedule-form-fields.js

```diff
@@ -0,0 +1,276 @@
+});
+
+export const activeField = () => ({
+  component: componentTypes.CHECKBOX,
+  name: 'enabled',
+  label: __('Active'),
+});
+
+export const actionField = (actionOptions, filterOptions, setData, data) => ({
+  component: componentTypes.SELECT,
+  id: 'action',
+  name: 'action_typ',
+  label: __('Action'),
+  initialValue: 'vm',
+  onChange: (value) => actionChange(value, filterOptions, setData, data),
+  isRequired: true,
+  options: restructureOptions(actionOptions),
+});
+
+export const filterField = (setData, data) => ({
+  component: componentTypes.SELECT,
+  id: 'filter_type',
+  name: 'filter_typ',
+  label: __('Filter'),
+  onChange: (value) => subActionChange(value, setData, data),
+  hideField: data.displayFields.hideFilterType,
+  options: data.options.subAction,
+  isRequired: true,
+});
+
+export const targetField = (data) => ({
+  component: componentTypes.SELECT,
+  id: 'filter_value',
+  name: 'filter_value',
+  label: __('Filter Item'),
+  hideField: data.displayFields.hideTarget,
+  options: data.options.target,
+  placeholder: __('<Choose>'),
+  includeEmpty: true,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideTarget }],
+});
+
+export const zoneField = (data) => ({
+  component: componentTypes.SELECT,
+  id: 'zone',
+  name: 'zone_id',
+  label: __('Zone'),
+  placeholder: __('<Choose>'),
+  includeEmpty: true,
+  hideField: data.displayFields.hideAutomationFields,
+  options: data.options.zone,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideAutomationFields }],
+});
+
+export const plainField = (name, text, data) => ({
+  component: 'plain-text',
+  name,
+  label: text,
+  element: 'h3',
+  hideField: data.displayFields.hideAutomationFields,
+});
+
+export const systemField = (data) => ({
+  component: componentTypes.SELECT,
+  id: 'system',
+  name: 'instance_name',
+  label: __('System/Proess'),
+  hideField: data.displayFields.hideAutomationFields,
+  options: data.options.request,
+  placeholder: __('<Choose>'),
+  includeEmpty: true,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideAutomationFields }],
+});
+
+export const objectMessageField = (data) => ({
+  component: componentTypes.TEXT_FIELD,
+  id: 'message',
+  name: 'object_message',
+  label: __('Message'),
+  hideField: data.displayFields.hideAutomationFields,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideAutomationFields }],
+});
+
+export const objectRequestField = (data) => ({
+  component: componentTypes.TEXT_FIELD,
+  id: 'request',
+  name: 'object_request',
+  label: __('Request'),
+  hideField: data.displayFields.hideAutomationFields,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideAutomationFields }],
+});
+
+export const objectTypeField = (setData, data) => ({
+  component: componentTypes.SELECT,
```

Comment:
Same Situation here as in the timezone select. There's a lot of options in this select and it would look better and be easier to navigate if it was searchable.

![image](https://user-images.githubusercontent.com/64800041/217946296-e19027dc-986e-4a27-837c-305c0be8c98e.png)

Reply:
added a search feature to the select box.